airflow/models/dag.py

```diff
@@ -451,6 +453,7 @@ def __init__(
             max_active_tasks = concurrency
         self._max_active_tasks = max_active_tasks
         self._pickle_id: int | None = None
+        self._display_name = dag_id if display_name is None else display_name
```

Comment:
This would allow the user to pass in an empty display name, which would look pretty terrible anywhere. I’d make the argument `display_name: str = ""` instead and do `self._display_name = display_name or dag_id` here.